src/collective/contact/widget/schema.py

```diff
@@ -10,13 +10,16 @@ class ContactList(RelationList):
     implements(IContactList)
     source_types = None
     review_state = None
+    prefilter_vocubulary = None
 
     def __init__(self, *args, **kwargs):
         self.addlink = kwargs.pop('addlink', True)
         self.source_types = kwargs.pop('source_types',
                                        self.source_types or None)
         self.review_state = kwargs.pop('review_state',
                                        self.review_state or None)
+        self.prefilter_vocubulary = kwargs.pop('prefilter_vocubulary',
```

Comment:
I need to use the context in the vocabulary, not available in a static vocabulary. Need to manage a named vocabulary or a source !

Reply:
I adapted the contact widget to use different kinds of vocabularies passed to _prefilter_vocubulary_:

- IVocabulary
- IVocabularyFactory (as named vocabulary)
- IContextSourceBinder